File: 1686/python3/main.py
class Solution:
    def stoneGameVI(self, aliceValues: list[int], bobValues: list[int]) -> int:
        n = len(aliceValues)
        commonValues = [0] * n
        for i in range(n):
            commonValues[i] = [
                aliceValues[i] + bobValues[i],
                aliceValues[i],
                bobValues[i],
            ]
        commonValues.sort(reverse=True)
        aliceScore = 0
        bobScore = 0
        for i in range(n):
            if i & 1 == 0:
                aliceScore += commonValues[i][1]
            else:
                bobScore += commonValues[i][2]
        return 0 if aliceScore == bobScore else 1 if aliceScore > bobScore else -1

File: 1686/go/main.go
func stoneGameVI(aliceValues []int, bobValues []int) int {
	scoreA, scoreB, n := 0, 0, len(aliceValues)
	commomValues := make([][]int, n)

	for i := 0; i < n; i++ {
		commomValues[i] = []int{aliceValues[i] + bobValues[i], aliceValues[i], bobValues[i]}
	}
	sort.Slice(commomValues, func(i int, j int) bool {
		return commomValues[i][0] > commomValues[j][0]
	})
	for i := 0; i < n; i++ {
		if (i & 1) == 0 {
			scoreA += commomValues[i][1]
		} else {
			scoreB += commomValues[i][2]
		}
	}
	if scoreA == scoreB {
		return 0
	} else if scoreA > scoreB {
		return 1
	} else {
		return -1
	}
}
